miasm2/ir/ir.py

```diff
@@ -256,15 +258,35 @@ def __init__(self, label, irs):
 
         assert isinstance(label, AsmLabel)
         self.label = label
-        self.irs = irs
-        self.except_automod = True
+        self._assignments = tuple(irs)
         self._dst = None
         self._dst_linenb = None
 
-    def _get_dst(self):
-        """Find the IRDst affectation and update dst, dst_linenb accordingly"""
-        if self._dst is not None:
-            return self._dst
+    @property
+    def irs(self):
+        return self._assignments
+
+    def reset_dst(self):
+        self._dst = None
+        self._dst_linenb = None
+
+    def is_dst_set(self):
+        return self._dst is not None
+
+    def add_assignblock(self, assignblock):
```

Comment:
The only API used in Miasm was this one. After a double check, it is only used in one file. So I am for removing it. What about you?
By the way the `__init__` is exactly the API you describe...